src/Language/Haskell/HGrep/Print.hs

```diff
@@ -74,14 +62,29 @@ printSearchResult (PrintOpts co lno) (SearchResult anns ast) =
     wholeSrc :: [Char]
     wholeSrc = EP.exactPrint ast anns
 
-    nill, src :: [[Char]]
-    (nill, src) = L.span null $ L.lines wholeSrc
+    nonEmptySrc :: [[Char]]
+    nonEmptySrc = L.filter (not . all isSpace) $ L.lines wholeSrc
```

Comment:
I think this `filter` can be avoided. ExactPrint doesn't seem to give us trailing spaces, so there's no need.

This line also seems to remove desired whitespace from the output before line numbers are added, so the numbers are wrong underneath empty lines.

```haskell
/tmp/Foo.hs:(3,1)-(7,5)
3  abc = do
4    def
5    -- This is line 6 in emacs
6    ghi
```

You could probably golf this patch down even further by just removing `nill` from the output :)

Reply:
Cool, I'll also go back to using `null` but that will mean any lines that simply have white space will be included (unless exact print turns those into empty strings as well)